homeassistant/components/proximity/coordinator.py

```diff
@@ -87,13 +96,55 @@ def __init__(
 
         self.state_change_data: StateChangedData | None = None
 
+    @callback
+    def async_add_entity_mapping(self, tracked_entity_id: str, entity_id: str) -> None:
+        """Add an tracked entity to proximity entity mapping."""
+        self.entity_mapping[tracked_entity_id].append(entity_id)
+
     async def async_check_proximity_state_change(
         self, entity: str, old_state: State | None, new_state: State | None
     ) -> None:
         """Fetch and process state change event."""
         self.state_change_data = StateChangedData(entity, old_state, new_state)
         await self.async_refresh()
 
+    async def async_check_tracked_entity_change(
+        self, event: EventType[er.EventEntityRegistryUpdatedData]
+    ) -> None:
+        """Fetch and process tracked entity change event."""
+        data = event.data
+        if data["action"] == "remove":
+            self._create_removed_tracked_entity_issue(data["entity_id"])
+
+        if data["action"] == "update" and "entity_id" in data["changes"]:
```

Comment:
Franck suggested we use the tracked entity registry entry id as unique id suffix instead of the entity id. The entry id won't change.

For tracked entities without a unique id we can still use the entity id, but it can't change so we don't need to handle that case.

Reply:
I guess/hope "tracked entity registry entry id" means the unique_id of the tracked entity 🙈 
if yes, that this should be ready -> 6e6e0835d4b5586f251afc5c0dc357485ae40783

Reply:
now if there is an entity registry entry for the tracked entity_id we use its `unique_id`, else we use the tracked entity_id
so no unique_id migration is needed anymore

Reply:
Nope, `entity_entry.id` not `entity_entry.unique_id`